extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentPusher.java

```diff
@@ -115,15 +118,14 @@ public DataSegment push(File inDir, DataSegment segment) throws IOException
           storageDir,
           segment.getShardSpec().getPartitionNum()
```

Comment:
it is a good idea but this will require that druid-api has to depend on hadoop APIs which i am afraid it will trigger push back from no-hadoop-fan-club.. most of the logic is encapsulated at `dataSegmentPusher.makeIndexPathName` so i guess we are ok.